packages/integration/src/lib/catalog/catalog-library-tool/catalog-library-tool.component.ts

```diff
@@ -92,4 +130,291 @@ export class CatalogLibraryToolComponent implements OnInit {
+          : 'https://tile.openstreetmap.org/{z}/{x}/{y}.png';
+        rv.so = osmSo;
+        break;
+      case 'wfs':
+        const wfsSo = so as WFSDataSourceOptions;
+        rv.layerName = wfsSo.params.featureTypes;
+        rv.url = wfsSo.url;
+        rv.so = wfsSo;
+        break;
+      case 'vector':
+        const featureSo = so as FeatureDataSourceOptions;
+        rv.layerName = '';
+        rv.url = featureSo.url;
+        rv.so = featureSo;
+        break;
+      case 'cluster':
+        const clusterSo = so as ClusterDataSourceOptions;
+        rv.layerName = '';
+        rv.url = clusterSo.url;
+        rv.so = clusterSo;
+        break;
+      case 'mvt':
+        const mvtSo = so as MVTDataSourceOptions;
+        rv.layerName = '';
+        rv.url = mvtSo.url;
+        rv.so = mvtSo;
+        break;
+      case 'carto':
+        const cartoSo = so as CartoDataSourceOptions;
+        rv.layerName = cartoSo.config.layers
+          .map((layer) => layer.options.sql)
+          .join(' ');
+        rv.url = `https://${cartoSo.account}.carto.com/api/v1/map`;
+        rv.so = cartoSo;
+        break;
+      default:
+        break;
+    }
+    if (rv.so) {
+      rv.id = generateIdFromSourceOptions(rv.so);
+      rv.url = rv.url?.startsWith('/')
+        ? window.location.origin + rv.url
+        : rv.url;
+    }
+
+    return rv;
+  }
+
+  getCatalogList(): void {
+    var catalogRank = 1;
+    const t = this.languageService.translate;
+    let bufferArray: CsvOutput[] = [
+      {
+        id: 'csvHeader',
+        rank: t.instant('igo.integration.catalog.csv.rank'),
+        layerTitle: t.instant('igo.integration.catalog.csv.layerTitle'),
+        layerGroup: t.instant('igo.integration.catalog.csv.layerGroup'),
+        catalog: t.instant('igo.integration.catalog.csv.catalog'),
+        provider: t.instant('igo.integration.catalog.csv.externalProvider'),
+        url: t.instant('igo.integration.catalog.csv.url'),
+        layerName: t.instant('igo.integration.catalog.csv.layerName'),
+        context: t.instant('igo.integration.catalog.csv.context'),
+        dataDescription: t.instant(
+          'igo.integration.catalog.csv.dataDescription'
+        )
+      }
+    ];
+    this.generatelist$$ = zip([
+      this.store.entities$.pipe(
+        switchMap((catalogs) => {
+          return forkJoin(
+            catalogs.map((catalog) =>
+              this.catalogService.loadCatalogItems(catalog).pipe(
+                map((loadedCatalogItems) => {
+                  return { catalog, loadedCatalogItems };
+                })
+              )
+            )
+          );
+        })
+      ),
+      this.contextService
+        .getLocalContexts()
+        .pipe(
+          switchMap((contextsList) =>
+            forkJoin(
+              contextsList.ours.map((context) =>
+                this.contextService.getLocalContext(context.uri)
+              )
+            )
+          )
+        )
+    ]).subscribe(
```

Comment:
Par rapport à mon commentaire sur la lisibilité et de décortiquer cette méthode... Par exemple, on pourrait isoler ce bout de code dans un méthode `getAllContexts` qui permettrait aux développeurs de comprendre rapidement ce que ça fait sans s'embourber sur le comment et la même logique pourrait être appliquer à d'autre section de cette méthode.
```suggestion
    this.getAllContexts().subscribe(
```